cardano-config/src/Cardano/Node/Configuration/Basics.hs

```diff
@@ -0,0 +1,66 @@
+-- | Basic types for configuration
+module Cardano.Node.Configuration.Basics
+  ( -- * FilePaths
+    FilePath (..)
+  , (</>)
+  , RelativeFilePath (..)
+  , anchorRelativePath
+
+    -- * Defaults
+  , Override (..)
+  , (.:=)
+  ) where
+
+import Control.Applicative ((<|>))
+import Data.Aeson
+import Data.Aeson.Types (Parser)
+import Data.Default
+import Data.String (IsString)
+import GHC.Generics (Generic)
+import GHC.TypeLits (Symbol)
+import qualified System.FilePath as F
+import Prelude hiding (FilePath)
+import qualified Prelude as P
+
+-- | A filepath annotated with the intended use for it
+newtype FilePath (s :: Symbol) = FilePath {getFilePath :: P.FilePath}
```

Comment:
The current `FilePath` conflicts with `System.FilePath.FilePath`. Using a different name would make it a bit unambiguous. What do you think about using a shorter name `File`? 
```suggestion
newtype File (s :: Symbol) = File {getFile :: P.FilePath}
```